### ISSUE TITLE
ESP32-S2 Support

### ISSUE THREAD

**Issue by moefear85** [glenn20/micropython-espnow-images#5]
Hi,

A precompiled image for the esp32 obviously isn't working on an esp32-s2. So I was wondering if there are plans to support it?

**Comment by glenn20** [comment]
Yeah - I have a few esp32-s2 boards on order. They should arrive soon and I can test them. Once they pass all the tests I'll add them to the image build (at least the GENERIC_S2 board build).

**Comment by moefear85** [comment]
ok thx...

On a similar note... I just tried compiling from your sources, for the esp32-wrover (4MB PSRAM). after flashing, it keeps resetting after the watchdog kicks in. I compiled with make BOARD=GENERIC_SPIRAM. Is this configuration currently supported?

**Comment by glenn20** [comment]
Thanks for the experience report. I don't have a wrover so haven't tested that - I'd have thought it unlikely there would be an issue. If you wish, you might like to try compiling from https://github.com/glenn20/micropython/tree/espnow-g20-v116. I regularly rebase the espnow-g20 branch against master to make it easier for review and hopefully merging someday. The espnow-g20-v116 branch is against the micropython v1.16 release which might be more stable.

My S2 devices (Feather S2s) are on the way ;-).

**Comment by moefear85** [comment]
unfortunately, it turns out (through further testing) vanilla micropython doesn't have working wifi for the esp32s2. so the problems seem to stem from there. I'm currently cloning the latest master with little hope that anything has changed since v1.16 release.

**Comment by glenn20** [comment]
Thanks for the report - I suspected as much. My S2s are delayed due to backlogs in mail delivery here in Aus.

**Comment by moefear85** [comment]
no prob... i'll keep checking in regular intervals for any news.... but as indicated, I think some heavy work needs to be done on the micropython side first before esp32s2 wifi will be usable (unless i'm doing something terribly wrong).

**Comment by glenn20** [comment]
Sorry for the long delay. After resolving a few issues I now have a working espnow for the ESP32S2. I have uploaded an ESP32S2 firmware image compiled against micropython v1.17 and ESP IDF v4.3.1.

The image passes all espnow tests in tests/multi_espnow/*.py.

I also had to implement a fix to limit the amount of RAM that micropython takes for it's heap memory (which left too little for the ESP networking and espnow memory buffers).

Let me know if you have any issues.
